gallery_dl/extractor/2chen.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for https://2chen.moe/"""
+
+from .common import Extractor, Message
+from .. import text
+
+
+class _2chenThreadExtractor(Extractor):
+    """Extractor for 2chen threads"""
+    category = "2chen"
+    subcategory = "thread"
+    directory_fmt = ("{category}", "{board}", "{thread} {title}")
+    filename_fmt = "{time} {filename}.{extension}"
+    archive_fmt = "{hash}"
+    root = "https://2chen.moe"
+    pattern = r"(?:https?://)?2chen\.moe/([^/?#]+)/(\d+)"
+    test = (
+        ("https://2chen.moe/jp/303786", {
+            "count": ">= 10",
+        }),
+    )
+
+    def __init__(self, match):
+        Extractor.__init__(self, match)
+        self.board, self.thread = match.groups()
+
+    def items(self):
+        url = "{}/{}/{}".format(self.root, self.board, self.thread)
+        page = self.request(url, encoding="utf-8").text
+        data = self.metadata(page)
+        yield Message.Directory, data
+        for post in self.posts(page):
+            if not post["url"]:
+                continue
+            post.update(data)
+            post["url"] = self.root + post["url"]
+            post["time"] = text.parse_int(post["date"].timestamp())
+            yield Message.Url, post["url"], text.nameext_from_url(
+                post["filename"], post)
+
+    def metadata(self, page):
+        board, pos = text.extract(page, 'class="board">/', '/<')
+        title = text.extract(page, "<h3>", "</h3>", pos)[0]
+        return {
+            "board" : board,
+            "thread": self.thread,
+            "title" : text.unescape(title),
+        }
+
+    def posts(self, page):
+        """Return iterable with relevant posts"""
+        return map(self.parse, text.extract_iter(
+            page, 'class="glass media', '</article>'))
+
+    def parse(self, post):
+        extr = text.extract_from(post)
+        return {
+            "name"    : extr("<span>", "</span>"),
```

Comment:
Sorry to bring this up now, but I think both `name` as well as `filename` further down need to be unescaped (`text.unescape()`)

Reply:
I'll update those (I haven't come across a name other than "Anonymous")
should I also `parse_int` the `no` ?

Reply:
> I haven't come across a name other than "Anonymous"

Right, but there is the possibility of someone using a custom name.
I guess it is not all that necessary for `name`, but at least `filename` should be unescaped since it is part of the default filename format siring.

> should I also `parse_int` the `no` ?

4chan has an actual number as `no`, so maybe.
I'd keep it as it is currently.
Type consistency is something for v2.0. Right now is doesn't really matter all that much.

Reply:
> Right, but there is the possibility of someone using a custom name.

Nevermind, custom names aren't a thing it seems.